sycl/test/devicelib/math_utils.hpp

```diff
@@ -1,25 +1,33 @@
 #ifndef MATH_UTILS
 #include <complex>
 #include <limits>
-using namespace std;
 // T must be float-point type
 template <typename T>
 bool is_about_FP(T x, T y) {
-  if (x == y)
-    return true;
+  bool ret;
+
+  // At least one input is nan
+  if (std::isnan(x) || std::isnan(y))
+    return std::isnan(x) && std::isnan(y);
+
+  // At least one input is inf
+  if (std::isinf(x) || std::isinf(y))
+    ret = (x == y);
+
+  // two finite
   else {
+    T threshold = std::numeric_limits<T>::epsilon() * 100;
     if (x != 0 && y != 0) {
-      T max_v = fmax(abs(x), abs(y));
-      return (abs(x - y) / max_v) <
-              numeric_limits<T>::epsilon() * 100;
-    }
-    else {
+      T max_v = std::fmax(std::abs(x), std::abs(y));
+      ret = (std::abs(x - y) / max_v) < threshold;
```

Comment:
```suggestion
      return std::abs(x - y) < threshold*max_v;
```
is more efficient and probably more in the spirit of what this function is_about.